Fix Hangul output in iOS and in Safari

Schahen opened the pull request:
Correct behaviour for Hangul input under certain circumstances 

Fixes [https://youtrack.jetbrains.com/issue/CMP-8773]

## Testing
`./gradlew testWeb`

## Release Notes
### Fixes - Web
 - Mobile. Composite input. When a syllable block is created, a new block is added instead of replacing the old one

Schahen committed:
Fix hangul output in iOS and in Safari

This fixes https://youtrack.jetbrains.com/issue/CMP-8773

Schahen committed:
Add comment that clarifies what's going on

eymar commented on compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/NativeInputEventsProcessor.kt:
```diff
@@ -146,6 +148,8 @@ internal abstract class NativeInputEventsProcessor(
                 if (deleteSize > 0) {
                     add(DeleteSurroundingTextCommand(deleteSize, 0))
+                } else if (deleteSize == 0) {
+                    add(BackspaceCommand())
```
As I remember from the logs you shared from iPad, only `deleteContentBackward` was dispatched there for this type of input.

Is it necessary to add this check (`deleteSize == 0`) here too? 

____

The name of the events suggests that `deleteContentBackward` can indeed have no selection - so just delete like a backspace would do.

But `insertReplacementText` hints that a selection must be present to replace something.

Schahen replied:
good catch! - this is fallacy of tuning something to test - (in this case, for `SafariCompositeInput::`input hangul-hol`) let me fix test instead

Schahen replied:
@eymar fixed Safari test and removed this snippet

Schahen committed:
correct SafariCompositeInput::`input hangul-hol` behviour

eymar reviewed (approved):
LGTM!

Schahen committed:
InputEvent::deleteContentBackwardSize is nullable

eymar commented on compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/NativeInputEventsProcessor.kt:
```diff
@@ -134,7 +134,7 @@ internal abstract class NativeInputEventsProcessor(
                     // The system first tells us to delete the old text,
                     // and then it would send the "insertText" event.
-                    val deleteSize = deleteContentBackwardSize
+                    val deleteSize = deleteContentBackwardSize ?: 0
```
in this case it will go to `else if` branch and execute Backspace command.

Schahen replied:
@eymar shouldn't we treat the null (deleteContentBackwarSize was not set at all) like zero?

Schahen committed:
Always set deleteContentBackwardSize

This leads to Removing Safari:: `input hangul-hol` test - unfortunately we can not change state of html input which is needed for this test to behave properly
Other possible solution would be not to set deleteContentBackwardSize if it somehow came from event already but this will mean we leak testing logic into application. This price would be too high

Schahen committed:
Correct name for CompositeInputTestSpec
